panel/src/components/Forms/Field/ObjectField.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+	<k-field v-bind="$props" class="k-object-field">
+		<!-- Remove button -->
+		<template v-if="!disabled" #options>
+			<k-button v-if="isEmpty" icon="add" @click="onAdd" />
+			<k-button v-else icon="remove" @click="onRemove" />
+		</template>
+
+		<table
+			v-if="!isEmpty"
+			:data-invalid="isInvalid"
+			class="k-table k-object-field-table"
+		>
+			<tbody>
+				<template v-for="field in fields">
+					<tr
+						v-if="$helper.field.isVisible(field, value)"
+						:key="field.name"
+						@click="open(field.name)"
+					>
+						<th data-mobile="true">
+							<button type="button">{{ field.label }}</button>
+						</th>
+						<k-table-cell
+							:column="field"
+							:field="field"
+							:mobile="true"
+							:value="value[field.name]"
+							@input="onCellInput(field.name, $event)"
+						/>
+					</tr>
+				</template>
+			</tbody>
+		</table>
+		<k-empty v-else :data-invalid="isInvalid" icon="box" @click="onAdd">
+			{{ empty || $t("field.object.empty") }}
+		</k-empty>
+
+		<k-form-drawer
+			ref="drawer"
+			v-bind="drawer"
+			@input="onDrawerInput"
+			@invalid="onErrors"
+		/>
+	</k-field>
+</template>
+
+<script>
+import { props as Field } from "../Field.vue";
+import { props as Input } from "../Input.vue";
+
+export default {
+	mixins: [Field, Input],
+	props: {
+		empty: String,
+		fields: Object,
+		value: Object
+	},
+	data() {
+		return {
+			object: this.value,
+			errors: []
+		};
+	},
+	computed: {
+		drawer() {
+			return {
+				icon: "box",
+				tab: "object",
+				tabs: {
+					object: {
+						fields: this.$helper.field.subfields(this, this.fields)
+					}
+				},
+				title: this.label,
+				value: this.object
+			};
+		},
+		isEmpty() {
+			if (!this.value) {
+				return true;
+			}
+
+			if (this.value && Object.keys(this.value).length === 0) {
+				return true;
+			}
+
+			return false;
+		},
+		isInvalid() {
+			// if field itself is required and empty
+			if (this.required === true && this.isEmpty) {
+				return true;
+			}
+
+			// if subfields are required but empty
+			for (const field in this.fields) {
+				if (
+					this.fields[field].required === true &&
+					this.$helper.object.isEmpty(this.value[field]) === true
```

Comment:
```suggestion
					this.$helper.object.isEmpty(this.value?.[field]) === true
```

This fixes the https://github.com/getkirby/kirby/pull/4640#issuecomment-1252015259 issue